### ISSUE TITLE
Quarrantine PII 

### ISSUE THREAD

**Issue by edsu** [edsu/anon#169]
As some might have [seen already](https://www.washingtonpost.com/politics/2018/09/28/fight-over-kavanaugh-nomination-finds-its-oddest-front-yet-wikipedia-pages/?utm_term=.de12bdda8084) congressedits has been suspended because it was tricked into tweeting Personally Identifying Information (PII) which is against Twitter (and Wikipedia's) Terms of Service.

After talking with [Daniel Schuman](https://twitter.com/danielschuman) I think it might be useful to introduce the ability to configure anon to filter out tweets that appear to contain PII, and to alert an admin via email or Twitter, who could then notify Wikipedia.

Given the imprecise nature of matching PII I think this code needs to live external to anon, and to be kept private to prevent obvious gaming of the logic. 

If this fails, then the options are to move to manual review of all edits (it's not a huge number), or discontinuing the bot, or at least my administering of it. However, I'm open to suggestions if people have other ideas about how best to proceed.

**Comment by antoinemcgrath** [comment]
The filtering idea seems necessary and worthwhile. Congressedits is an appreciated and important bot for transparency. 

Line 30 of the [congresseditors.coffee]( https://github.com/edsu/congresseditors/blob/f7040b1a4233787ef5600ac32b92a857f06de5a5/congresseditors.coffee) could be a good place to add the PII check. 
Regarding notifying wiki I didn't see a PII report form, it looks like they have a pref. [to be emailed](https://en.wikipedia.org/wiki/Wikipedia:Contact_us_-_Readers). It would make sense to have twit direct message a responder who can evaluate the wiki post. Or the responsibility could be crowd sourced, twit could simply publicly post notice that a wiki edit is believed to have PII and encourage followers to investigate and report.

**Comment by edsu** [comment]
Alas, it looks like Twitter is not open to reopening the congressedits account. Of course someone else could easily run the bot themselves since the code is right here. But if I do that it would be interpreted by Twitter as trying to get around the rules, and could compromise some of my other accounts. All that said I do think it would be useful to have this PII quarantine behavior built into anon.

![img_9360](https://user-images.githubusercontent.com/33829/46498601-e3ecf780-c7eb-11e8-8ae4-3027e5611333.PNG)

**Comment by edsu** [comment]
@antoinemcgrath thanks for finding that info about how to notify Wikipedia. I like the idea of DM'ing the admin of the bot to alert Wikipedia. I worry that a public post could, in the case where there are many thousands of followers, still lead some to want to use it as a megaphone.

Btw, the CoffeeScript code has been converted to JavaScript.

**Comment by JoshData** [comment]
This is unfortunate. We should find some Twitter public policy people to connect with.

**Comment by edsu** [comment]
I would like to get the word out to current admins of the other anon bots that are still alive, and have lots of followers. The pattern could repeat elsewhere.

**Comment by dphiffer** [comment]
Yeah detecting PII is a Hard Problem, especially if somebody is determined to intentionally coerce a bot into tweeting it out. Maybe it would help to enumerate which things you'd want to detect?

* Phone numbers
* Addresses
* Latitude/Longitude coordinates
* Other things?
